to-json-schema.js

```diff
@@ -21,17 +21,52 @@ const typeMappings = {
   uuid: 'string',
 };
 
+const commonAttributesMapping = (avroDefinition, jsonSchema) => {
+  if (avroDefinition.doc) jsonSchema.description = avroDefinition.doc;
+  if (avroDefinition.default !== undefined) jsonSchema.default = avroDefinition.default;
+};
+
+const exampleAttributeMapping = (typeInput, example, jsonSchemaInput) => {
+  let type = typeInput;
+  let jsonSchema = jsonSchemaInput;
+
+  // Map example to first non-null type
+  if (Array.isArray(typeInput) && typeInput.length > 0) {
+    const pickSecondType = typeInput.length > 1 && typeInput[0] === 'null';
+    type = typeInput[+pickSecondType];
+    jsonSchema = jsonSchema.oneOf[0];
+  }
+  
+  if (example === undefined || jsonSchema.examples || Array.isArray(type)) return;
+
+  switch (type) {
+  case 'boolean':
+    jsonSchema.examples = [example === 'true'];
+    break;
+  case 'int':
+    jsonSchema.examples = [parseInt(example, 10)];
+    break;
+  default:
+    jsonSchema.examples = [example];
+  }
+};
+
 module.exports.avroToJsonSchema = async function avroToJsonSchema(avroDefinition) {
   const jsonSchema = {};
   const isUnion = Array.isArray(avroDefinition);
 
   if (isUnion) {
     jsonSchema.oneOf = [];
+    let nullDef = null;
     for (const avroDef of avroDefinition) {
       const def = await avroToJsonSchema(avroDef);
-      jsonSchema.oneOf.push(def);
+      const defType = avroDef.type || avroDef;
```

Comment:
```suggestion
      // avroDef can be { type: 'int', default: 1 } and this is why avroDef.type has priority here
      const defType = avroDef.type || avroDef;
```

sorry but I need such comments in code as it is hard to maintain such project overtime, where changes are done very rarely